### PR TITLE
LoanField + NRIC

### PR DESCRIPTION
**Pull request by FongYuan**
Contains a base class called LoanField, and a new data field for Loan called Nric.

Planning to redo this PR, please do not approve.

### CODE REVIEW

File: src/main/java/seedu/address/model/loan/Nric.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * Represents a Loan's NRIC number in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValid}
+ */
+public class Nric extends LoanField<String> {
+
+    public static final String MESSAGE_CONSTRAINTS =
+        "NRIC numbers should start with a S, T, F or G, followed by 7 decimal digits, then end with a letter.";
+
+    public static final Predicate<String> isValid = test ->
+        test.matches("[STFGstfg]\\d{7}[A-Za-z]");
```
Comment (prokarius) [inline src/main/java/seedu/address/model/loan/Nric.java]: Last letter is in `[A-JZa-jz]`.

Comment (prokarius) [reply src/main/java/seedu/address/model/loan/Nric.java]: Propose change to `[STFGstfg]\d{7}[A-JZa-jz]`

Comment (prokarius) [reply src/main/java/seedu/address/model/loan/Nric.java]: Also, Singapore NRIC's last digit is a checksum. We should implement that check sum as well.

**Comment by FongYuan** [comment]
I will redo the PR and get the DataField class up. I'll leave NRIC for someone else to implement.

**Pull request closed without merging**